atom/browser/api/atom_api_session.cc

```diff
@@ -443,9 +443,12 @@ void Session::ClearStorageData(mate::Arguments* args) {
     // https://w3c.github.io/mediacapture-main/#dom-mediadeviceinfo-deviceid
     MediaDeviceIDSalt::Reset(browser_context()->prefs());
   }
-  storage_partition->ClearData(options.storage_types, options.quota_types,
-                               options.origin, base::Time(), base::Time::Max(),
-                               base::Bind(&OnClearStorageDataDone, callback));
+
+  storage_partition->ClearData(
+      options.storage_types, options.quota_types, options.origin, base::Time(),
+      base::Time::Max(),
+      base::Bind(&ResolvePromise, util::CopyablePromise(promise)));
```

Comment:
why not move the promise here?

Reply:
@miniak `ClearData` only accepts a `RepeatingCallback` with `base::Bind`, and Promises are move-only so we can't move it and thus need to use CopyablePromises

Reply:
gotcha